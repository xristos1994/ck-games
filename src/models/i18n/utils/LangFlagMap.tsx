import React, { FC, ReactElement } from "react";
import { GreekFlagIcon, UkFlagIcon } from "@components/icons";
import { availableLangs } from "./availableLangs";

interface IProps {
  langCode: string;
}
const LangFlagMap: FC<IProps> = ({ langCode }): ReactElement => {
  switch (langCode) {
    case availableLangs.en.code:
      return <UkFlagIcon />;
    case availableLangs.el.code:
      return <GreekFlagIcon />;
    default:
      return <span>langCode</span>;
  }
};

export { LangFlagMap };
