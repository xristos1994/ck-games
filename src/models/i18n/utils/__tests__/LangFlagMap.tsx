import React from 'react';
import { LangFlagMap } from './../LangFlagMap';
import renderer from 'react-test-renderer';

describe('models/i18n/utils//LangFlagMap', () => {
  it('renders correctly', () => {
    let tree = renderer.create(<LangFlagMap langCode="en" />).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(<LangFlagMap langCode="el" />).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(<LangFlagMap langCode="notExistingCode" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
