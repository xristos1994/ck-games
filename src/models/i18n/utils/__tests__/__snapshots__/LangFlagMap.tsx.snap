// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/i18n/utils//LangFlagMap renders correctly 1`] = `
<svg
  height="100%"
  id="Layer_1"
  version="1.1"
  viewBox="0 0 512 512"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="256"
    cy="256"
    fill="#F0F0F0"
    r="256"
  />
  <g>
    <path
      d="M52.92,100.142c-20.109,26.163-35.272,56.318-44.101,89.077h133.178L52.92,100.142z"
      fill="#0052B4"
    />
    <path
      d="M503.181,189.219c-8.829-32.758-23.993-62.913-44.101-89.076l-89.075,89.076H503.181z"
      fill="#0052B4"
    />
    <path
      d="M8.819,322.784c8.83,32.758,23.993,62.913,44.101,89.075l89.074-89.075L8.819,322.784L8.819,322.784 z"
      fill="#0052B4"
    />
    <path
      d="M411.858,52.921c-26.163-20.109-56.317-35.272-89.076-44.102v133.177L411.858,52.921z"
      fill="#0052B4"
    />
    <path
      d="M100.142,459.079c26.163,20.109,56.318,35.272,89.076,44.102V370.005L100.142,459.079z"
      fill="#0052B4"
    />
    <path
      d="M189.217,8.819c-32.758,8.83-62.913,23.993-89.075,44.101l89.075,89.075V8.819z"
      fill="#0052B4"
    />
    <path
      d="M322.783,503.181c32.758-8.83,62.913-23.993,89.075-44.101l-89.075-89.075V503.181z"
      fill="#0052B4"
    />
    <path
      d="M370.005,322.784l89.075,89.076c20.108-26.162,35.272-56.318,44.101-89.076H370.005z"
      fill="#0052B4"
    />
  </g>
  <g>
    <path
      d="M509.833,222.609h-220.44h-0.001V2.167C278.461,0.744,267.317,0,256,0 c-11.319,0-22.461,0.744-33.391,2.167v220.44v0.001H2.167C0.744,233.539,0,244.683,0,256c0,11.319,0.744,22.461,2.167,33.391 h220.44h0.001v220.442C233.539,511.256,244.681,512,256,512c11.317,0,22.461-0.743,33.391-2.167v-220.44v-0.001h220.442 C511.256,278.461,512,267.319,512,256C512,244.683,511.256,233.539,509.833,222.609z"
      fill="#D80027"
    />
    <path
      d="M322.783,322.784L322.783,322.784L437.019,437.02c5.254-5.252,10.266-10.743,15.048-16.435 l-97.802-97.802h-31.482V322.784z"
      fill="#D80027"
    />
    <path
      d="M189.217,322.784h-0.002L74.98,437.019c5.252,5.254,10.743,10.266,16.435,15.048l97.802-97.804 V322.784z"
      fill="#D80027"
    />
    <path
      d="M189.217,189.219v-0.002L74.981,74.98c-5.254,5.252-10.266,10.743-15.048,16.435l97.803,97.803 H189.217z"
      fill="#D80027"
    />
    <path
      d="M322.783,189.219L322.783,189.219L437.02,74.981c-5.252-5.254-10.743-10.266-16.435-15.047 l-97.802,97.803V189.219z"
      fill="#D80027"
    />
  </g>
</svg>
`;

exports[`models/i18n/utils//LangFlagMap renders correctly 2`] = `
<svg
  height="100%"
  version="1.1"
  viewBox="0 0 512 512"
  width="100%"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="256"
    cy="256"
    fill="#F0F0F0"
    r="256"
  />
  <g>
    <path
      d="M256,189.217h247.181c-6.419-23.814-16.175-46.255-28.755-66.783H256V189.217z"
      fill="#338AF3"
    />
    <path
      d="M96.643,456.348h318.713c23.363-18.608,43.399-41.21,59.069-66.783H37.574 C53.245,415.137,73.281,437.74,96.643,456.348z"
      fill="#338AF3"
    />
    <path
      d="M89.043,61.939c-20.126,17.331-37.525,37.74-51.469,60.496h51.469V61.939z"
      fill="#338AF3"
    />
    <path
      d="M256,256c0-26.076,0-47.899,0-66.783H155.826V256H89.043v-66.783H8.819 C3.08,210.511,0,232.893,0,256s3.08,45.489,8.819,66.783h494.363C508.92,301.489,512,279.107,512,256H256z"
      fill="#338AF3"
    />
    <path
      d="M256,0c-35.549,0-69.404,7.253-100.174,20.348v102.087H256c0-23.312,0-43.139,0-66.783h159.357 C371.626,20.824,316.249,0,256,0z"
      fill="#338AF3"
    />
  </g>
</svg>
`;

exports[`models/i18n/utils//LangFlagMap renders correctly 3`] = `
<span>
  notExistingCode
</span>
`;
